arch-x86: Improve KVM set XCR

abmerop opened the pull request:
This adds two failsafes which may cause a panic on some machines. First, check the host machine has the KVM XCR capability before calling getXCRs or setXCRs. Second, ensure the x87 bit, which must always be one, will always return at least one by modifying the return value in readMiscReg.

Change-Id: I5e778acc926a47443ef6cef29fabd84eb69bb9ba

mattsinc reviewed (approved):
@v-ramadas : FYI

powerjg reviewed (dismissed):
I am blocking this change until we reconcile it with https://github.com/gem5/gem5/pull/1127

There have been multiple KVM changes and some combination has broken KVM on some machines. We need to sort this out before we make more changes to KVM.

If this fixes #1126, that would be great. Please let us know.

cc: @andysan @nmosier @Harshil2107

abmerop committed:
arch-x86: Improve KVM set XCR

This adds two failsafes which may cause a panic on some machines. First,
check the host machine has the KVM XCR capability before calling getXCRs
or setXCRs. Second, ensure the x87 bit, which must always be one, will
always return at least one by modifying the return value in readMiscReg.

Change-Id: I5e778acc926a47443ef6cef29fabd84eb69bb9ba
